File: Two Pointers/125. Valid Palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        String f=s.toLowerCase();
        int n= f.length();
        ArrayList<Character> list = new ArrayList<>();
        for(int i=0; i<n; i++){
            if(Character.isLetterOrDigit(f.charAt(i))){
                list.add(f.charAt(i));
            }
        }
        int size=list.size();
        int l=0, u=size-1;
        while(l<=u){
            if(list.get(l).equals(list.get(u))){
                l++; 
                u--;
            }
            else return false;
        }
        return true;
    }
}

File: Two Pointers/125. Valid Palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        arr=[]
        for ch in s:
            if ch.isalnum():
                chl=ch.lower()
                arr.append(chl)
        n=len(arr)
        i=0
        j=n-1
        while i<=j:
            if arr[i]==arr[j]:
                i+=1
                j-=1
            else:
                return False
        return True

        